framework/ui/view/qmltooltip.cpp

```diff
@@ -0,0 +1,108 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#include "qmltooltip.h"
+
+#include <QWidget>
+#include <QToolTip>
+#include <QGuiApplication>
+#include <QStyleHints>
+#include <QCursor>
+#include <QQuickWindow>
+
+#include "log.h"
+
+using namespace mu::framework;
+
+QmlToolTip::QmlToolTip(QObject* parent)
+    : QObject(parent)
+{
+    m_timer.setSingleShot(true);
+    connect(&m_timer, &QTimer::timeout, this, &QmlToolTip::doShowToolTip);
+}
+
+void QmlToolTip::show(QQuickItem* item, const QString& text)
+{
+    if (item != m_item) {
+        if (m_item) {
+            disconnect(m_item, &QObject::destroyed, this, &QmlToolTip::doHide);
+        }
+
+        m_item = item;
+        m_text = text;
+
+        if (m_item) {
+            connect(m_item, &QObject::destroyed, this, &QmlToolTip::doHide);
+
+            const int interval = item ? qApp->styleHints()->mousePressAndHoldInterval() : 100;
+            m_timer.start(interval);
+        } else {
+            doHide();
```

Comment:
It's not really obvious that a method called a "show" can also "hide" something
Maybe "toggle" would be a more convenient name for the method

Reply:
what he is hiding is more likely an exception handling rather than normal behaviour.